feat: disable omni_submitUserOpTest in production worker

silva-fj opened the pull request:
## Summary

Added `test-endpoints` feature flag to control test-only RPC methods in omni-executor. The `omni_submitUserOpTest` endpoint is now excluded from production builds by default and only included when explicitly enabled with the feature flag.

silva-fj committed:
feat: add test-endpoints feature flag to control test RPC methods

silva-fj committed:
chore: update build configs to use test-endpoints feature

Kailai-Wang commented on tee-worker/omni-executor/Makefile:
```diff
@@ -22,6 +22,11 @@ $(BIN_NAME):
 	cargo build --release --locked --features=$(COMPILATION_FEATURES)
 	cp target/release/executor-worker $(BIN_NAME)
 
+.PHONY: $(BIN_NAME)-test
+$(BIN_NAME)-test:
+	cargo build --release --locked --features=$(COMPILATION_FEATURES),test-endpoints
+	cp target/release/executor-worker $(BIN_NAME)
```
May I know how should I make build the staging worker now?

So far it's been `make SGX=1 SGX_DEBUG=0`

silva-fj replied:
We can do: `make omni-executor-test SGX=1 SGX_DEBUG=0`

Kailai-Wang replied:
> We can do: `make omni-executor-test SGX=1 SGX_DEBUG=0`

But it won't build sgx metadata files like `omni-executor.manifest.sgx` / `omni-executor.sig` etc, right?

silva-fj replied:
Right. I've change it so we can do: `make SGX=1 SGX_DEBUG=0 TEST_BUILD=1` now

kziemianek replied:
This means we no logner can deploy releases to staging :(

Kailai-Wang replied:
On staging we deploy it manually anyway 😂 

I'm fine with `make SGX=1 SGX_DEBUG=0 TEST_BUILD=1`

silva-fj committed:
feat: add TEST_BUILD flag for conditional test-endpoints feature

- Add TEST_BUILD variable to control test-endpoints feature inclusion
- Remove standalone omni-executor-test target
- Unified build flow now handles both production and test builds with SGX

silva-fj committed:
Merge branch 'dev' into p-1696-disable-omni_submituseroptest-in-prod-worker
